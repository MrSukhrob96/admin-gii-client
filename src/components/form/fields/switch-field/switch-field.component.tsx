import { hoc } from '@utils/hoc';
import { useSwitchFieldProps } from './switch-field.props';
import { SwitchFieldContainer } from './switch-field.style';

/**
 * <SwitchField />
 */
export const SwitchField = hoc(useSwitchFieldProps, () => (
  <SwitchFieldContainer>
    {/* content */}
    {/* content */}
    {/* content */}
  </SwitchFieldContainer>
));
