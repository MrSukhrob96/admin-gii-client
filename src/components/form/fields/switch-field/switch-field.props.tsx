import { ReactNode } from "react";
import { Control, ControllerProps, useController } from "react-hook-form";

/**
 * <SwitchField /> props
 */
export type SwitchFieldProps = {
  name: string;
  helperText?: string;
  label?: ReactNode;
  rules?: ControllerProps['rules'];
  control: Control<any>;
}

export const useSwitchFieldProps = (props?: SwitchFieldProps) => {
  const {
    field: { value, onChange },
    fieldState: { error }
  } = useController({
    name: props.name,
    rules: props.rules,
    control: props.control
  });

  return { value, error, onChange }
};

