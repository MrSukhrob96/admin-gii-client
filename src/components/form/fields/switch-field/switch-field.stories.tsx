import { ComponentStory, ComponentMeta } from '@storybook/react';
import { SwitchField } from './switch-field.component';
import '@styles/index.css';

export default {
  title: 'Common/SwitchField',
  component: SwitchField,
} as ComponentMeta<typeof SwitchField>

const Template: ComponentStory<typeof SwitchField> = (args) => <SwitchField {...args} />;

export const SwitchFieldInitial = Template.bind({});
SwitchFieldInitial.args = {
}