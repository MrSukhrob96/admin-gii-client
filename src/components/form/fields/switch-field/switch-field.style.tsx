import styled from 'styled-components'

export const SwitchFieldContainer = styled.div`

`