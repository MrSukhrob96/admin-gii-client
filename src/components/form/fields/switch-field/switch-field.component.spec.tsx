import { customRender } from '@test-utils';
import { renderHook } from '@testing-library/react-hooks';
import { useForm } from 'react-hook-form';
import { SwitchField } from './switch-field.component';

describe('SwitchField', () => {
  test('Should render SwitchField component', async () => {
    const { result } = renderHook(() => useForm({
      defaultValues: { test: '' }
    }));
    const { asFragment } = customRender(
      <SwitchField name='test' control={result.current.control} label='Switch' />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});