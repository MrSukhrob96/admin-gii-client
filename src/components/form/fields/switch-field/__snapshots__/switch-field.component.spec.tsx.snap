// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchField Should render SwitchField component 1`] = `
<DocumentFragment>
  <div
    class="bp4-form-group"
  >
    <div
      class="bp4-form-content"
    >
      <label
        class="bp4-control bp4-switch bp4-large"
      >
        <input
          id="test-input"
          type="checkbox"
        />
        <span
          class="bp4-control-indicator"
        />
        Switch
      </label>
    </div>
  </div>
</DocumentFragment>
`;
